### PR TITLE
add ci verify via github actions

### PR DESCRIPTION
**Pull request by mrparkers**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  push:
+    branches:
+      - "*"
+  pull_request:
+    branches: [main]
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
```
Comment (gesparza3) [inline .github/workflows/test.yml:13]: Overall I like the changes a lot, moving these commands to a Makefile makes it look a lot cleaner...I see that you moved these steps out of a dedicated workflow container. Is that mainly for simplification?

Comment (mrparkers) [reply .github/workflows/test.yml:13]: Nope, I just copied what was in liatrio/rode-collector-sonarqube and made a few changes.  I'll work on getting this to run in a container, I don't think I've done that before with gh actions.

File: Makefile [hunk 2]
```diff
@@ -8,3 +9,15 @@ tools:
 generate:
 	docker build ./scripts/generate -t ghcr.io/liatrio/rode-api-generate:latest
 	docker run -it --rm -v $$(pwd):/rode-api ghcr.io/liatrio/rode-api-generate:latest
+
+fmtcheck:
+	lineCount=$(shell gofmt -l -s $(GOFMT_FILES) | wc -l | tr -d ' ') && exit $$lineCount
```
Comment (aalsabag) [inline Makefile:14]: exiting with the number of formatting errors. That's a good one

**Review by aalsabag — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (mrparkers) [reply Makefile:14]: stole it from https://github.com/mrparkers/terraform-provider-keycloak/blob/master/makefile :smile:

**Pull request merged**